### PR TITLE
Add mild domain randomization to Car Racing Env

### PR DESCRIPTION
**Pull request by jjshoots**
# Description

This adds a mild domain randomization trait to the Car Racing Env.

Simply pass `domain_randomize=True`, and the road and background colours will be randomized.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### Before 
before1

### After
after1
after2
after3

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by pseudo-rnd-thoughts** [comment]
Could you add a test for `hardcore=True`?

**Comment by jjshoots** [comment]
> 

Yea I couldn't find any tests for CarRacing, and I agree with you that there should be. I'll get to implementing one then.

File: gym/envs/box2d/car_racing.py [hunk 1]
```diff
@@ -183,6 +193,22 @@ def _destroy(self):
...
+            self.norm_road_color = self.np_random.uniform(0, 0.8, size=3)
+
+            self.bg_color = self.np_random.uniform(0, 210, size=3)
+
+            self.grass_color = np.copy(self.bg_color)
+            idx = self.np_random.integers(3)
+            self.grass_color[idx] += 20
+        else:
+            # default colours
+            self.norm_road_color = np.array([0.4, 0.4, 0.4])
```
Comment (RedTachyon) [inline gym/envs/box2d/car_racing.py:208]: Do we have to store the colors with two different representations? (one is 0-1, the other is 0-255)

File: gym/envs/box2d/car_racing.py [hunk 2]
```diff
@@ -128,10 +125,15 @@ class CarRacing(gym.Env, EzPickle):
...
-    There are no arguments supported in constructing the environment.
+    `lap_complete_percent` dictates the percentage of tiles that must be visited by
+    the agent before a lap is considered complete.
+
+    Passing `hardcore=True` enabled the domain randomized variant of the environment.
+    In this scenario, the background and track colours are different on every reset.
 
     ### Version History
-    - v0: Current version
+    - v1: Current version (0.23.1)
```
Comment (RedTachyon) [inline gym/envs/box2d/car_racing.py:135]: I guess v0 can be labelled as the original version; v1 will come live with 0.24.0 at the earliest, so probably better to put that (seeing as 0.23.1 is already out)

**Review by RedTachyon — COMMENTED** [review]
Overall looks good, just left one nitpicky comment for code quality

Actually, one thought. Would it make sense to name this something else than `hardcore`? I know there are some envs that already follow this naming pattern, but it's extraordinarily undescriptive.

Also - we might want to register an extra env to make it easier to create this version, something like "CarRacingHardcore" or whatever Hardcore might get renamed to.

File: gym/envs/box2d/car_racing.py (hunk 1 above)
Comment (jjshoots) [reply gym/envs/box2d/car_racing.py]: Gotcha, fixed it

**Comment by jjshoots** [comment]
@RedTachyon I've changed `hardcore` to `domain_randomize`, hopefully that's more descriptive, also registered it.

**Pull request merged**